src/MSBuild/XMake.cs

```diff
@@ -1086,7 +1089,7 @@ ProfilerLogger profilerLogger
 
                     // Propagate the profiler flag into the project load settings so the evaluator
                     // can pick it up
-                    if (profilerLogger != null)
+                    if (profilerLogger != null || enableProfiler)
```

Comment:
>(profilerLogger != null || enableProfile [](start = 23, length = 40)

shouldn't this be 'enableProfiler'  directly?

Reply:
I didn't do this since it really should conceptually be either. If the logger is attached (somehow) it needs to be set. Or if we want to set it explicitly via the other mechanism.